src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -53,11 +57,54 @@ namespace Microsoft.VisualStudio.LanguageServices.Setup
         termValues: new[] { Guids.CSharpProjectExistsInWorkspaceUIContextString, Guids.VisualBasicProjectExistsInWorkspaceUIContextString })]
     internal class RoslynPackage : AbstractPackage
     {
-        private VisualStudioWorkspace _workspace;
-        private IComponentModel _componentModel;
-        private RuleSetEventHandler _ruleSetEventHandler;
-        private ColorSchemeApplier _colorSchemeApplier;
-        private IDisposable _solutionEventMonitor;
+        // The randomly-generated key name is used for serializing the ILSpy decompiler EULA preference to the .SUO
```

Comment:
why save to SUO, instead of like the user profile?

seems unfortunate that if i flip to different projects (or make a new solution/prj) i would have to accept the prompt again, right?

Reply:
This is the approval we got from legal for persisting the option.